File: akka-essentials/Chapter03/PingPongScala/src/main/scala/PingPongApp.scala
package akka.examples

import akka.actor.{ActorSystem, Props}

object PingPongApp {

  def main(args: Array[String]): Unit = {
    val _system = ActorSystem("PingPong")
    val actor = _system.actorOf(Props[PingPongActor](), "pingPongActor")
    actor ! PING

    terminate(_system)
  }

  private def terminate(system: ActorSystem): Unit =
    try {
      Thread.sleep(2000)
      println(">>> Press ENTER to exit <<<")
      System.in.read()
    }
    catch {
      case _: Exception => /* ignored */
    }
    finally {
      system.terminate()
    }

}

File: akka-essentials/Chapter03/PingPongBuilder/src/main/java/PingPongApp.java
package akka.examples;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;

public class PingPongApp {
    
    public static void main(String[] args) throws Exception {
        ActorSystem _system = ActorSystem.create("PingPong");
        ActorRef actor = _system.actorOf(
            Props.create(PingPongActor.class),
            "pingPongActor"
        );
        actor.tell(PingPongActor.PING, ActorRef.noSender());

        terminate(_system);
    }

    private static void terminate(ActorSystem system) {
        try {
            Thread.sleep(2000);
            System.out.println(">>> Press ENTER to exit <<<");
            System.in.read();
        }
        catch (Exception e) {
            /* ignored */
        }
        finally {
            system.terminate();
        }
    }

}
